dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -29,6 +29,82 @@ def self.extract_parent_emails
     ContactRollupsV2.execute_query_in_transaction(query)
   end
 
+  def self.extract_scripts_taught
+    source_sql = <<~SQL
+      SELECT u.email, sc.name AS script_name, c.name AS course_name, MAX(se.updated_at) AS updated_at
+      FROM sections AS se
+      JOIN users AS u ON se.user_id = u.id
+      JOIN scripts AS sc ON sc.id = se.script_id
+      LEFT JOIN course_scripts AS cs ON cs.script_id = se.script_id
+      LEFT JOIN courses AS c ON c.id = cs.course_id
+      WHERE u.email > ''
+      GROUP BY u.email, sc.name, c.name
+    SQL
+
+    query = get_extraction_query(source_sql, 'email', ['script_name', 'course_name'], true, 'dashboard.sections')
+    ContactRollupsV2.execute_query_in_transaction(query)
+  end
+
+  def self.extract_courses_taught
+    source_sql = <<~SQL
+      SELECT u.email, courses.name AS course_name, MAX(se.updated_at) AS updated_at
+      FROM users AS u
+      JOIN sections AS se ON se.user_id = u.id
+      JOIN courses ON courses.id = se.course_id
+      WHERE u.email > ''
+      GROUP BY u.email, courses.name
+    SQL
+
+    query = get_extraction_query(source_sql, 'email', ['course_name'], true, 'dashboard.sections')
+    ContactRollupsV2.execute_query_in_transaction(query)
+  end
+
+  def self.extract_professional_learning_attendance_old_attendance_model
+    # "section_type" is not null only in cases where sections was used for
+    # PD attendance. It's always null when a section represents an actual
+    # classroom of students (the vast majority of rows in the sections table).
+    source_sql = <<~SQL
+      SELECT u.email, se.section_type, MAX(GREATEST(se.updated_at, f.updated_at)) AS updated_at
+        FROM users AS u
```

Comment:
nit: is this indentation intentional? Same for query in extract_professional_learning_attendance_new_attendance_model.

Reply:
Not really sure what I was thinking here, fixed for consistency.